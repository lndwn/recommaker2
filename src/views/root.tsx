import * as React from 'react'
import { connect, ConnectedProps } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { defaultTheme, darkTheme } from 'theme'
import { useSystemColorScheme, useDocumentTitle, usePointerInfo } from 'utils'
import { GlobalStyle } from 'style/global-style.sc'
import { Grid } from 'components'
import { Controls } from './fragments/controls'
import { Sidebar } from 'components/sidebar'
import { ThemeSwitcher } from './fragments/theme-switcher'
import { Dropdown } from './fragments/dropdown'
import { AppState } from 'store/root-reducer'

const mapState = (state: AppState) => ({
  colorScheme: state.colorScheme.overrideScheme,
})
const connector = connect(mapState)
type RootProps = ConnectedProps<typeof connector>

const Root = (props: RootProps) => {
  useDocumentTitle('Root â€” RSK')
  usePointerInfo()

  const systemScheme = useSystemColorScheme()
  console.group('Color Scheme')
  console.info('User color scheme', props.colorScheme)
  console.info('System color scheme', systemScheme)
  const theme =
    props.colorScheme === 'dark'
      ? darkTheme
      : props.colorScheme === 'light'
      ? defaultTheme
      : systemScheme === 'dark'
      ? darkTheme
      : defaultTheme
  console.info(theme)
  console.groupEnd()

  return (
    <ThemeProvider theme={theme}>
      <Sidebar />
      <Grid
        gridTemplateColumns="repeat(auto-fill, minmax(320px, 1fr))"
        p="5"
        gridGap="5"
        alignItems="start"
        justifyItems="stretch">
        <ThemeSwitcher />
        <Controls />
        <Dropdown />
      </Grid>
      <GlobalStyle />
    </ThemeProvider>
  )
}

const ConnectedRoot = connector(Root)
export { ConnectedRoot as Root }
