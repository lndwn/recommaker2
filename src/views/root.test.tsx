import * as React from 'react'
import { render } from '@testing-library/react'
import { Root } from './root'

describe('root', () => {
  it('renders with theme context', () => {
    const { container, getByText } = render(<Root />)

    expect(getByText(/hello world/i))
    expect(container.firstChild).toMatchSnapshot()
  })
  it.todo('renders in dark mode')
  it.todo('renders in light mode')
  it.todo('switches between light and dark mode')
  it.todo('renders svg icons')
  it.todo('sets the document title')
})
