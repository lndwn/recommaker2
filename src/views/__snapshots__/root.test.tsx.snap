// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root renders with theme context 1`] = `
.c4 {
  margin-right: 1ch;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100vw;
  height: 100vh;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
  color: inherit;
}

@media (prefers-reduced-motion:reduce) {

}

<div
  class="c0 c1"
  height="100vh"
  width="100vw"
>
  <div
    class="c2"
  >
    <div
      class="c3"
      color="inherit"
      font-size="4"
    >
      <span />
    </div>
    <div
      class="c4"
    />
    Hello World
  </div>
</div>
`;
